src/writer/statement.ts

```diff
@@ -0,0 +1,31 @@
+import { getAddress } from '@ethersproject/address';
+import db from '../helpers/mysql';
+import { getSpace } from '../helpers/actions';
+
+export async function verify(msg): Promise<any> {
+  const space = await getSpace(msg.space);
+
+  if (!space) return Promise.reject('invalid space');
+
+  if (msg.about.length > 140) return Promise.reject('about is too long');
+
+  return true;
+}
+
+export async function action(msg, ipfs, receipt, id): Promise<void> {
+  const item = {
+    id,
```

Comment:
This id is based on the entire data? 
`const id = snapshot.utils.getHash(body.data);` 

So it can update from time to time? can we create an id based on `delegate` and `space` so we can update the record on duplicate

Reply:
The primary key is delegate + space

Reply:
Cool. but `id` and `ipfs` are always the first received ones. not updated

Reply:
maybe can leave `id` but will update `ipfs`